contrib/globalsign/mgo/mgo.go

```diff
@@ -0,0 +1,395 @@
+package mgo
+
+import (
+	"context"
+
+	"github.com/globalsign/mgo"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+type mongoConfig struct {
+	serviceName string
+}
+
+// MongoOption represents an option that can be passed to Dial
+type MongoOption func(*mongoConfig)
+
+func defaults(cfg *mongoConfig) {
+	cfg.serviceName = "mongodb"
+}
+
+// WithServiceName sets the service name for a given MongoDB context.
+func WithServiceName(name string) MongoOption {
+	return func(cfg *mongoConfig) {
+		cfg.serviceName = name
+	}
+}
+
+// Dial opens a connection to a MongoDB server and ties it to the
+// given context for tracing MongoDB calls.
+func Dial(ctx context.Context, url string, opts ...MongoOption) (*WrapSession, error) {
+	session, err := mgo.Dial(url)
+	s := &WrapSession{
+		Session: session,
+		ctx:     ctx,
+	}
+
+	defaults(&s.cfg)
+	for _, fn := range opts {
+		fn(&s.cfg)
+	}
+
+	return s, err
+}
+
+// WrapSession wraps a Session with the Context and Config information
+// needed to support tracing.
+type WrapSession struct {
```

Comment:
Can we name these `Session`, `Database`, etc instead of `WrapSession`? `WrapX` sounds like it should be a method

Reply:
Sure.

Reply:
Done